modules/oe_theme_helper/src/EventSubscriber/NodeMetadataEventSubscriber.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\EventSubscriber;
+
+use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\node\NodeInterface;
+use Drupal\oe_theme_helper\Event\NodeMetadataEvent;
+use Drupal\oe_theme_helper\MetadataEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Subscribes to the event fired on try to retrieve node for metadata.
+ */
+class NodeMetadataEventSubscriber implements EventSubscriberInterface {
```

Comment:
Actually, thinking about it, let's call this one `DefaultNodeMetadataEventSubscriber` because it is the default implementation. Meaning that it returns the node for the default use cases that the theme actually ships with.